src/lightning/fabric/connector.py

```diff
@@ -220,10 +226,23 @@ def _check_config_and_set_final_flags(
 
         self._accelerator_flag = accelerator
 
-        supported_precision = get_args(_PRECISION_INPUT_STR) + get_args(_PRECISION_INPUT_INT)
+        supported_precision = (
+            get_args(_PRECISION_INPUT_STR) + get_args(_PRECISION_INPUT_INT) + get_args(_PRECISION_INPUT_STR_LEGACY)
+        )
         if precision not in supported_precision:
             raise ValueError(f"Precision {repr(precision)} is invalid. Allowed precision values: {supported_precision}")
-        self._precision_input = cast(_PRECISION_INPUT_STR, str(precision))
+
+        precision = str(precision)  # convert int flags to str here to enable the legacy-conversion below
+
+        if precision in get_args(_PRECISION_INPUT_STR_LEGACY):
+            if str(precision)[:2] not in ("32", "64"):
+                rank_zero_warn(
+                    f"{precision} is supported for historical reasons but its usage is discouraged. "
+                    f"Please set your precision to {_PRECISION_INPUT_STR_LEGACY_CONVERSION[precision]} instead!"
+                )
+            precision = _PRECISION_INPUT_STR_LEGACY_CONVERSION[precision]
+
+        self._precision_input = cast(_PRECISION_INPUT_STR, precision)
```

Comment:
would this block perhaps fit into a utility function, just to keep the size of _check_config_and_set_final_flags contained?

Reply:
can do!